sdks/python/apache_beam/coders/coders.py

```diff
@@ -878,6 +877,27 @@ def __hash__(self):
     common_urns.INTERVAL_WINDOW_CODER, IntervalWindowCoder)
 
 
+class PaneInfoCoder(FastCoder):
+  """Coder for an PaneInfo descriptor."""
+
+  def _create_impl(self):
+    return coder_impl.PaneInfoCoderImpl()
+
+  def is_deterministic(self):
+    return True
+
+  def as_cloud_object(self):
+    raise NotImplementedError(
+        'The PaneInfoCoder should not be explicitly constructed in a pipeline '
+        'graph.')
+
+  def __eq__(self, other):
+    return type(self) == type(other)
+
+  def __hash__(self):
+    return hash(type(self))
+
+
```

Comment:
<!--thread_id:cc_173305213_t; commit:aa1b931750aa409e1d26e603fccd0fd82b9484da; resolved:1-->
<!--section:context-quote-->
> **robertwb** wrote:
> Rather than introducing a PaneInfoCoder, just instantiate the PaneInfoCoderImpl in WindowedValueCoderImpl's constructor. (There's a TODO to remove the parameterizablility of the TimestampCoder as well.) Actually, we could consider just making {en,de}code_pane_info methods on WindowedValueCoder itself.

<!--section:body-->
I am going to get rid of PaneInfoCoder but keep the PaneInfoCoderImpl, because it's more natural to keep it separate, as it would clutter the WindowedValueCoderImpl with encode/decode_paneinfo_to_stream and estimate_paneinfo_size.